core/src/util/fastmap.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <algorithm>
+
+namespace Tangram {
+
+// see also:
+// http://www.boost.org/doc/libs/1_59_0/doc/html/boost/container/flat_map.html
+// https://realm.io/assets/news/binary-search/blog.cpp
+
+template<typename K, typename T>
+struct fastmap {
+    std::vector<std::pair<K, T>> map;
```

Comment:
If we're getting picky about performance, do you think separate vectors of keys and values would provide better cache-locality for the binary search?